src/validatePlanPerformance.ts

```diff
@@ -0,0 +1,73 @@
+import WebSiteManagementClient from "azure-arm-website";
+import { AppServicePlan } from "azure-arm-website/lib/models";
+import { ConfigurationTarget, MessageItem, workspace, WorkspaceConfiguration } from "vscode";
+import { AppServicePlanNameStep, IAppServiceWizardContext } from "vscode-azureappservice";
+import { createAzureClient, DialogResponses, IActionContext, UserCancelledError } from "vscode-azureextensionui";
+import { AppServiceDialogResponses, extensionPrefix, maxNumberOfSites } from "./constants";
+import { ext } from "./extensionVariables";
+import { nonNullProp } from "./utils/nonNull";
+
+export async function validatePlanPerformance(wizardContext: IAppServiceWizardContext, rgName: string, newPlanName: string): Promise<void> {
```

Comment:
Based on my other comments, I'm beginning to think we should rename this to `setDefaultRgAndPlanName` so that this code can be included in the function:

```
// this should always be set when in the basic creation scenario
const location: Location = nonNullProp(wizardContext, 'location');	            const location: Location = nonNullProp(wizardContext, 'location');
wizardContext.newResourceGroupName = `appsvc_rg_${wizardContext.newSiteOS}_${location.name}`;	            wizardContext.newResourceGroupName = `appsvc_rg_${wizardContext.newSiteOS}_${location.name}`;
wizardContext.newPlanName = `appsvc_asp_${wizardContext.newSiteOS}_${location.name}`;	            wizardContext.newPlanName = `appsvc_asp_${wizardContext.newSiteOS}_${location.name}`;
```

It just feels like a cleaner delineation than `validatePlanPerformance` because it's not obvious _why_ you're validating the plan performance (to pick the default names).